sorting/pancake_sort.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file
+ * @brief pancake sort sorts a disordered stack of pancakes by flipping any number of pancakes using a spatula using minimum number of flips.
+ *
+ * @details
+ * Unlike a traditional sorting algorithm, which attempts to sort with the fewest comparisons possible,
+ * the goal is to sort the sequence in as few reversals as possible.
+ * Overall time complexity of pancake sort is O(n^2)
+ * For example: example 1:-
+ * Disordered pancake sizes: {2,5,3,7,8} 
+ * Sorted: {2,3,5,7,8}
+ * For example: example 2:-
+ * Disordered pancake sizes: {22,51,37,73,81} 
+ * Sorted: {22,37,51,73,81}
+ * @author [Divyansh Gupta](https://github.com/divyansh12323)
+ * @see more on [Pancake sort](https://en.wikipedia.org/wiki/Pancake_sorting)
+ * @see related problem [Leetcode](https://leetcode.com/problems/pancake-sorting/)
+*/
+
+#include <iostream>   // for io operations
+#include <vector>     // for std::vector
+#include <algorithm>  // for std::is_sorted
+#include <cassert>    // for std::assert
+
+/**
+ * @namespace sorting
+ * Sorting algorithms
+ */
+namespace sorting {
+    /**
+    * This implementation is for a C-style array input that gets modified in place.
+    * @param [start,end] arr our vector of elements.
+    * @param n size of given array
+    */
+    template <typename T>
+    void reverse(std::vector <T>& arr, int start, int end) { 
+	    T temp; 
+	    while (start <= end) { 
+		    temp = arr[start]; 
+		    arr[start] = arr[end]; 
+		    arr[end] = temp; 
+		    start++; 
+		    end--; 
+	    } 
+    } 
+
+    template <typename T>
+    int pancakeSort(std::vector <T>& arr, int n)  {
+        for (int i=n; i>1; --i) {
+            int mi = 0, j; //intialize some variables.
+            T maxE = 0;
+            for(j=0; j<i; j++) {
+                if(arr[j] >= maxE) {
+                    maxE = arr[j];
+	                mi = j;
+	            }
+	        }
+		    if (mi != i-1) //check for reversing 
+			 { 
+			    reverse(arr, 0, mi); 
+			    reverse(arr, 0, i-1); 
+		    } 
+	    } 
+    } 
```

Comment:
@divyansh12323 the return type is int you haven't returned anything.
That's why one of the checks is failing

Reply:
got it , i'll change